packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1460,6 +1591,51 @@ export class Bucket extends BucketBase {
 
     this.accessControl = BucketAccessControl.LOG_DELIVERY_WRITE;
   }
+
+  private parseInventoryConfiguration(): CfnBucket.InventoryConfigurationProperty[] | undefined {
+    if (!this.inventories || this.inventories.length === 0) {
+      return undefined;
+    }
+
+    this.inventories.forEach(inventory =>
+      (inventory.destination.addToResourcePolicy(new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ['s3:PutObject'],
+        resources: [
+          inventory.destination.bucketArn,
+          inventory.destination.arnForObjects('*'),
+        ],
+        principals: [new iam.ServicePrincipal('s3.amazonaws.com')],
+        conditions: {
+          ArnLike: {
+            'aws:SourceArn': this.bucketArn,
```

Comment:
This is just to limit the permissions on the source of the bucket.
https://docs.aws.amazon.com/AmazonS3/latest/dev/example-bucket-policies.html#example-bucket-policies-use-case-9

I've added this just for having a more scoped down policy, what you think?
I think we could also add the SourceAccount condition whenever the bucketOwner is given.

Reply:
> I've added this just for having a more scoped down policy, what you think?

👍 

> I think we could also add the SourceAccount condition whenever the bucketOwner is given.

Im honestly not entirely sure what the purpose of the owner id is. Is it to validate that both buckets reside in the same account?

Reply:
Yes, since a bucket name can be reused. S3 added this to validate that the bucket is still owned by the expected party before they write the reports into the inventory bucket.

Reply:
> I think we could also add the SourceAccount condition whenever the bucketOwner is given.

scratch that, I was confused :)

SourceAccount would be the current stack account? but this is me assuming it would be a good idea

Reply:
@iliapolo I'm having second thoughts of including this permissions in this function, would it make sense to make it an API instead or remove them at all?

Reply:
I don't think this API is currently warranted, and it definitely should be done implicitly by us and not by the user.
Why don't you want to include it? Im ok with the permission not being too fancy, but we need to add the minimum policies to make the feature work all around.

Reply:
I think my concern is around imported buckets, which are not neceserily managed by the stack or the current account.
This would mean we can't add the permissions to it, but maybe I'm wrong.

I've seen that severAccessLogs are doing something similar and is checking the bucket by using the `instanceof` to differentiate a Bucket from an IBucket.

```ts
    if (props.serverAccessLogsBucket instanceof Bucket) {
      props.serverAccessLogsBucket.allowLogDelivery();
    }
```

Then this way we can add the permissions only when the bucket is an instance of the class